### PR TITLE
Fix richTextLabel scroll showing

### LINKED ISSUE #61324: Editor Log doesn't activate scroll immediately when needed
### Godot version

a76db34

### System information

Windows 10 x64

### Issue description

godot windows tools 64_Gmkk8PKIX4

### Steps to reproduce

1. Print something that's longer than Output box height
2. You can't scroll
3. The scroll appears when you switch to another tab and back

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by iwek7**
This should fix #61324. The problem was that `_process_line_caches` function was only processing items that were newly added to  RitchTextLabel. So only if whole cache was invalidated or needed to be resized (for instance where switching windows) size of all lines was taken into account. Now for purpose of checking if scroll should appear or disappear we take into account also offset of last not invalidated line.

Small demo:
output_scroll

Worth noting that exactly same issue plagued regular RichTextLabel node, this fixes both of them.

### CODE REVIEW

File: scene/gui/rich_text_label.cpp [hunk 1]
```diff
@@ -1576,6 +1576,10 @@ int RichTextLabel::_find_first_line(int p_from, int p_to, int p_vofs) const {
 	return l;
 }
 
+float RichTextLabel::_calculate_line_vertical_offset(const RichTextLabel::Line &line) const {
```
Comment (bruvzg) [inline scene/gui/rich_text_label.cpp:1579]:
```suggestion
_FORCE_INLINE_ float RichTextLabel::_calculate_line_vertical_offset(const RichTextLabel::Line &line) const {
```

File: scene/gui/rich_text_label.h [hunk 2]
```diff
@@ -117,6 +117,10 @@ class RichTextLabel : public Control {
 		int char_count = 0;
 
 		Line() { text_buf.instantiate(); }
+
+		float get_height(float line_separation) const {
```
Comment (bruvzg) [inline scene/gui/rich_text_label.h:121]:
```suggestion
		_FORCE_INLINE_ float get_height(float line_separation) const {
```

File: scene/gui/rich_text_label.h [hunk 3]
```diff
@@ -504,6 +508,8 @@ class RichTextLabel : public Control {
 	void _scroll_changed(double);
 	int _find_first_line(int p_from, int p_to, int p_vofs) const;
 
+	float _calculate_line_vertical_offset(const Line& line) const;
```
Comment (bruvzg) [inline scene/gui/rich_text_label.h:511]:
```suggestion
	_FORCE_INLINE_ float _calculate_line_vertical_offset(const Line& line) const;
```

**Review by bruvzg — APPROVED** [review]

**Comment by iwek7** [comment]
I squashed commits and (hopefully) fixed formatting.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**